components/match2/wikis/arenaofvalor/matchgroup_input_custom.lua

```diff
@@ -0,0 +1,610 @@
+					match = matchFunctions.getPlayersOfTeam(match, opponentIndex, opponent.name, opponent.players)
+				end
+			elseif opponent.type == 'solo' then
+				opponent.match2players = Json.parseIfString(opponent.match2players) or {}
+				opponent.match2players[1].name = opponent.name
+			else
+				error('Unsupported Opponent Type "' .. (opponent.type or '') .. '"')
+			end
+
+			opponents[opponentIndex] = opponent
+		end
+	end
+
+	--apply walkover input
+	match.walkover = string.upper(match.walkover or '')
+	if Logic.isNumeric(match.walkover) then
+		local winnerIndex = tonumber(match.walkover)
+		opponents = matchFunctions._makeAllOpponentsLoseByWalkover(opponents, _STATUS_DEFAULT_LOSS)
+		opponents[winnerIndex].status = _STATUS_DEFAULT_WIN
+		match.finished = true
+	elseif Logic.isNumeric(match.winner) and Table.includes(_ALLOWED_STATUSES, match.walkover) then
+		local winnerIndex = tonumber(match.winner)
+		opponents = matchFunctions._makeAllOpponentsLoseByWalkover(opponents, match.walkover)
+		opponents[winnerIndex].status = _STATUS_DEFAULT_WIN
+		match.finished = true
+	end
+
+	-- see if match should actually be finished if bestof limit was reached
+	if isScoreSet and not Logic.readBool(match.finished) then
+		local firstTo = math.ceil(match.bestof/2)
+		for _, item in pairs(opponents) do
+			if tonumber(item.score or 0) >= firstTo then
+				match.finished = true
+				break
+			end
+		end
+	end
+
+	-- see if match should actually be finished if score is set
+	if isScoreSet and not Logic.readBool(match.finished) and match.hasDate then
+		local currentUnixTime = os.time(os.date('!*t'))
+		local lang = mw.getContentLanguage()
+		local matchUnixTime = tonumber(lang:formatDate('U', match.date))
+		local threshold = match.dateexact and _SECONDS_UNTIL_FINISHED_EXACT
+			or _SECONDS_UNTIL_FINISHED_NOT_EXACT
+		if matchUnixTime + threshold < currentUnixTime then
+			match.finished = true
+		end
+	end
+
+	-- apply placements and winner if finshed
+	if
+		not String.isEmpty(match.winner) or
+		Logic.readBool(match.finished) or
+		CustomMatchGroupInput.placementCheckSpecialStatus(opponents)
+	then
+		match.finished = true
+		match, opponents = CustomMatchGroupInput.getResultTypeAndWinner(match, opponents)
+	end
+
+	-- Update all opponents with new values
+	for opponentIndex, opponent in pairs(opponents) do
+		match['opponent' .. opponentIndex] = opponent
+	end
+	return match
+end
+
+function matchFunctions._makeAllOpponentsLoseByWalkover(opponents, walkoverType)
+	for index, _ in pairs(opponents) do
+		opponents[index].score = _NOT_PLAYED_SCORE
+		opponents[index].status = walkoverType
+	end
+end
+
+-- Get Playerdata from Vars (get's set in TeamCards)
+function matchFunctions.getPlayersOfTeam(match, oppIndex, teamName, playersData)
+	-- match._storePlayers will break after the first empty player. let's make sure we don't leave any gaps.
+	playersData = Json.parseIfString(playersData) or {}
+	local players = {}
+	local count = 1
```

Comment:
This counter seems unused

Reply:
What should I changed/add/removed with?

I attempt to remove either 465-468 entirely for 467-468 got me this 
![image](https://user-images.githubusercontent.com/88981446/149673438-bc1dcc72-b9f0-4cae-9e39-69b20f42d89f.png)

Reply:
Only 468 & 485

Reply:
> Only 468 & 485

It returns me this error 

608: 'end' expected (to close 'function' at line 464) near '<eof>'.
(608 = **return CustomMatchGroupInput** )

Reply:
Sounds like you're removing an `end`, should be removing `count = count + 1`